server/src/domain/media/media.service.ts

```diff
@@ -81,6 +79,63 @@ export class MediaService {
     return true;
   }
 
+  async handleQueueMigration() {
+    const assetPagination = usePagination(JOBS_ASSET_PAGINATION_SIZE, (pagination) =>
+      this.assetRepository.getAll(pagination),
+    );
+
+    for await (const assets of assetPagination) {
+      for (const asset of assets) {
+        await this.jobRepository.queue({ name: JobName.MIGRATE_ASSET, data: { id: asset.id } });
+      }
+    }
+
+    const people = await this.personRepository.getAll();
+
+    for (const person of people) {
+      await this.jobRepository.queue({
+        name: JobName.MIGRATE_PERSON,
+        data: { id: person.id },
+      });
+    }
+
+    while ((await this.jobRepository.getJobCounts(QueueName.MIGRATION)).waiting > 1) {
```

Comment:
Can you help explain this while block?

Reply:
I think the idea is to poll for when all the other jobs complete so the queue job is the only one left. When this is the case, it means migration is complete and it can remove the other directories. 

But I'm not sure if this check actually has this behavior. A job could be active, paused, etc. but not waiting, so it seems like it could break out of this while loop before the other jobs are finished. And this job would be active so if there's only one waiting job, it would be a job that should complete before this code moves on.

Reply:
@mertalev is right.
The idea was to avoid deleting directories that have just been created by another job and not yet filled with new files.
And I feel like the paused jobs actually don't make a difference, because if paused, it is also fine to delete all empty folders. In that case I won't interfere with a running case, deleting directories that have just been created.

About the active jobs however, I think you're right and I will need to wait until there are no active jobs as well.

Nevertheless, it might very well be that I'm a) completely overlooking something or that there is b), a much better way to do this. In both cases feedback would be much appreciated!

Reply:
What do you think about having a cronjob to run at a particular time of the day to delete empty directories when there are no jobs active?

Reply:
Hm, I'm not sure... It feels like unexpected behavior. I believe we should keep the folder structure untouched as long as not explicitly told to migrate/cleanup something. Maybe others would like to comment on this?

Reply:
Honestly probably not that important for thumbnail directories. I am fine running it at the beginning of the job though.